tools/licenses/lib/main.dart

```diff
@@ -37,132 +33,117 @@ abstract class _RepositoryEntry implements Comparable<_RepositoryEntry> {
 }
 
 abstract class _RepositoryFile extends _RepositoryEntry {
-  _RepositoryFile(_RepositoryDirectory parent, fs.File io) : super(parent, io);
-
-  Iterable<License>? get licenses;
-
-  @override
-  String get libraryName => parent!.libraryName;
+  _RepositoryFile(_RepositoryDirectory super.parent, fs.File super.io);
 
   fs.File get ioFile => super.io as fs.File;
 }
 
 abstract class _RepositoryLicensedFile extends _RepositoryFile {
-  _RepositoryLicensedFile(_RepositoryDirectory parent, fs.File io) : super(parent, io);
-
-  // file names that we are confident won't be included in the final build product
-  static final RegExp _readmeNamePattern = RegExp(r'\b_*(?:readme|contributing|patents)_*\b', caseSensitive: false);
-  static final RegExp _buildTimePattern = RegExp(r'^(?!.*gen$)(?:CMakeLists\.txt|(?:pkgdata)?Makefile(?:\.inc)?(?:\.am|\.in|)|configure(?:\.ac|\.in)?|config\.(?:sub|guess)|.+\.m4|install-sh|.+\.sh|.+\.bat|.+\.pyc?|.+\.pl|icu-configure|.+\.gypi?|.*\.gni?|.+\.mk|.+\.cmake|.+\.gradle|.+\.yaml|pubspec\.lock|\.packages|vms_make\.com|pom\.xml|\.project|source\.properties|.+\.obj|.+\.autopkg|Brewfile)$', caseSensitive: false);
-  static final RegExp _docsPattern = RegExp(r'^(?:INSTALL|NEWS|OWNERS|AUTHORS|ChangeLog(?:\.rst|\.[0-9]+)?|.+\.txt|.+\.md|.+\.log|.+\.css|.+\.1|doxygen\.config|Doxyfile|.+\.spec(?:\.in)?)$', caseSensitive: false);
-  static final RegExp _devPattern = RegExp(r'^(?:codereview\.settings|.+\.~|.+\.~[0-9]+~|\.clang-format|swift\.swiftformat|\.gitattributes|\.landmines|\.DS_Store|\.travis\.yml|\.cirrus\.yml|\.cache|\.mailmap|CODEOWNERS|TESTOWNERS)$', caseSensitive: false);
-  static final RegExp _testsPattern = RegExp(r'^(?:tj(?:bench|example)test\.(?:java\.)?in|example\.c)$', caseSensitive: false);
-  // The ICU library has sample code that will never get linked.
-  static final RegExp _icuSamplesPattern = RegExp(r'.*(?:icu\/source\/samples).*$', caseSensitive: false);
-
-  bool get isIncludedInBuildProducts {
-    return !io.name.contains(_readmeNamePattern)
-        && !io.name.contains(_buildTimePattern)
-        && !io.name.contains(_docsPattern)
-        && !io.name.contains(_devPattern)
-        && !io.name.contains(_testsPattern)
-        && !io.toString().contains(_icuSamplesPattern)
-        && !isShellScript;
-  }
-
-  bool get isShellScript => false;
+  _RepositoryLicensedFile(super.parent, super.io);
+
+  License extractInternalLicense() {
```

Comment:
These two methods should definitely have declaration comments explaining what they are supposed to do, so implementers understand their roles.

Reply:
done